ctx.go

```diff
@@ -1029,6 +1030,73 @@ func (c *DefaultCtx) QueryFloat(key string, defaultValue ...float64) float64 {
 	return value
 }
 
+// QueryParser parses a query parameter value from the given context and returns it as the specified type.
+// The function supports parsing integers, booleans, floats, and strings.
+//
+// Default to empty or invalid key is 0.
+//
+//	GET /?name=alex&wanna_cake=2&id=
+//	QueryParser[int]("wanna_cake", 1) == 2
+//	QueryParser[int]("name", 1) == 1
+//	QueryParser[int]("id", 1) == 1
+//	QueryParser[int]("id") == 0
+//
+// Default to empty or invalid key is true.
+//
+//	Get /?name=alex&want_pizza=false&id=
+//	QueryParser[bool]("want_pizza") == false
+//	QueryParser[bool]("want_pizza", true) == false
+//	QueryParser[bool]("name") == false
+//	QueryParser[bool]("name", true) == true
+//	QueryParser[bool]("id") == false
+//	QueryParser[bool]("id", true) == true
+//
+// Default to empty or invalid key is 0.
+//
+//	GET /?name=alex&amount=32.23&id=
+//	QueryParser[float64]("amount") = 32.23
+//	QueryParser[float64]("amount", 3) = 32.23
+//	QueryParser[float64]("name", 1) = 1
+//	QueryParser[float64]("name") = 0
+//	QueryParser[float64]("id", 3) = 3
+//
+// If the generic type is not written explicitly, the function will try to infer the type from the default value.
+// If the default value is not provided, the function will return the zero value of the type.
+//
+//	GET /?name=alex&wanna_cake=2&id=
+//	QueryParser("wanna_cake", 1) == 2
+func QueryParser[T any](c Ctx, key string, defaultValue ...T) T {
```

Comment:
1. could you rename it to just Query 
we always use parsers when we assign several data to a struct, but here we only want to access one value

2. we can use a contrains from the supported types instead of any (string | bool | float64 | .... )